types/network.go

```diff
@@ -29,6 +30,10 @@ type ConnectionConfig struct { //nolint:govet // fieldalignment
 	BearerToken string
 	// UserAgent is the User-Agent header sent to the Prometheus server.
 	UserAgent string
+	// ProtobufMessage is the Prometheus protobuf message to send.
+	ProtobufMessage promconfig.RemoteWriteProtoMsg
+	// MetadataCacheSize is the size of the LRU cache used for tracking Metadata to support sparse metadata sending
```

Comment:
Should this callout that it's only valid when used with Prometheus Remote Write v2?